src/service/invoice-service.ts

```diff
@@ -137,97 +165,72 @@ export default class InvoiceService {
       transfer: invoice.transfer ? TransferService.asTransferResponse(invoice.transfer) : undefined,
       description: invoice.description,
       pdf: invoice.pdf ? invoice.pdf.downloadName : undefined,
-      currentState: InvoiceService.asInvoiceStatusResponse(invoice.latestStatus),
+      currentState: InvoiceService.asInvoiceStatusResponse(invoice.invoiceStatus[invoice.invoiceStatus.length - 1]),
```

Comment:
I'm not sure whether we should assume that the last invoice status in the list is the most recent one. Maybe write a small lambda function to pick the newest?

Reply:
fixed